projects/solar/src/relay_fsm.c

```diff
@@ -59,3 +59,8 @@ StatusCode relay_fsm_close(void) {
 StatusCode relay_fsm_open(void) {
   return event_raise_priority(RELAY_EVENT_PRIORITY, SOLAR_RELAY_EVENT_OPEN, 0);
 }
+
+void relay_err_cb(const GpioAddress *address, void *context) {
+  LOG_DEBUG("DRV120 RELAY ERR TRIGGERED\n");
+  fault_handler_raise_fault(EE_SOLAR_FAULT_DRV120, 0);
+}
```

Comment:
Instead of having this be a public function, I would suggest moving the drv120 initialization to `relay_fsm_init` and making this function static. (I think this makes sense because relay_fsm "owns" the drv120 in some sense.)

Also, `fault_handler_raise_fault` already logs, so you don't have to here.